crates/bevy_render/src/render_task/extract.rs

```diff
@@ -0,0 +1,27 @@
+use super::RenderTask;
+use crate::{sync_world::RenderEntity, Extract};
+use bevy_camera::Camera;
+use bevy_ecs::system::{Commands, Query};
+
+// TODO: Use SyncComponentPlugin or ExtractComponentPlugin?
```

Comment:
I'm 90% sure that `ExtractComponentPlugin` handles everything you're trying to do here. Including handling when a Camera is removed.

Maybe not mutating `T` in the main world, but then again. I'm not quite sure of all the things you're trying to do with `RenderTask`. If you have more precise questions, I think I can answer them though.

Reply:
Yeah I think you're right. And I think I can get away with not needing to handle main world mutations if I move temporal reset tracking to a dedicated component.